lib/Semantic_check.mli

```diff
@@ -12,3 +12,9 @@ val check_that_all_functions_have_definition : bool ref
 val model_name : string ref
 (** A reference to hold the model name. Relevant for checking variable
     clashes and used in code generation. *)
+
+val inferred_unsizedtype_of_indexed :
```

Comment:
Could you add a docstring here?

Reply:
Yep. I wasn't sure this was the right function so was waiting to run it by you and get it all okayed.

Reply:
This is definitely the function I was thinking of. It seems right to me.